### PR TITLE
Pax 143 add device commands

### PR DESCRIPTION
**Pull request by includeamin**
# changes

- [x] add device CURD command

### CODE REVIEW

File: paxexpress_cli/devices/core.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    DeviceCreateResponseModel,
+    DeviceDeleteResponseModel,
+    DeviceListResponseModel,
+    DeviceModel,
+    DeviceProvisioningModel,
+)
+import httpx
+
+
+def add_device():
```
Comment (beshoyabdelmalak) [inline paxexpress_cli/devices/core.py:22]: Return Type annotations should be added

File: paxexpress_cli/devices/models.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class DeviceListResponseModel(BaseModel):
+    data: List[DeviceModel]
+
+
+class DeviceResponseModel(BaseModel):
+    data: DeviceModel
+
+
+class DeviceCreateResponseModel(BaseModel):
```
Comment (beshoyabdelmalak) [inline paxexpress_cli/devices/models.py:52]: You can just use the `GenericResponseModel` from authentication and delete this model and `DeviceDeleteResponseModel` as they are all the same

File: paxexpress_cli/devices/core.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    body = pydantic_to_prompt(model=CreateDeviceModel)
+    response = httpx.post(url=url, json=body.dict(), headers=header)
+    response_handler(response=response, return_model=DeviceCreateResponseModel)
+
+
+def update_device():
+    username, header = get_auth_header_and_username()
+    if not username:
+        return
+    url = get_url()
```
Comment (beshoyabdelmalak) [inline paxexpress_cli/devices/core.py:36]: this function is not yet completed

Comment (includeamin) [reply paxexpress_cli/devices/core.py:36]: you are right! 👍

File: paxexpress_cli/devices/models.py (hunk 2 above)
Comment (includeamin) [reply paxexpress_cli/devices/models.py:52]: I think it is better that each API has its own response model (despite the similarity).

File: paxexpress_cli/utils.py [hunk 4]
```diff
@@ -137,11 +140,17 @@ def pydantic_to_prompt(model: ClassVar) -> Any:
 
 
 def custom_prompt(**kwargs):
+    regex = kwargs.pop("regex")
```
Comment (includeamin) [inline paxexpress_cli/utils.py:143]:
```suggestion
    regex = kwargs.pop("regex", None)
```

File: paxexpress_cli/utils.py [hunk 5]
```diff
@@ -137,11 +140,17 @@ def pydantic_to_prompt(model: ClassVar) -> Any:
...
+    regex_error_message = kwargs.pop("regex_error_message")
```
Comment (includeamin) [inline paxexpress_cli/utils.py:144]:
```suggestion
    regex_error_message = kwargs.pop("regex_error_message", None)
```

File: paxexpress_cli/utils.py (hunk 4 above)
Comment (beshoyabdelmalak) [reply paxexpress_cli/utils.py:143]: You are right

**Pull request merged**